pull_chart_tarball.go

```diff
@@ -37,7 +37,7 @@ func (c *Client) doFile(ctx context.Context, req *http.Request) (string, error)
 	o := func() error {
 		resp, err := c.httpClient.Do(req)
 		if err != nil {
-			return microerror.Mask(err)
+			return backoff.Permanent(microerror.Mask(err))
```

Comment:
Cancelling retry if there is no such host. 
e.g. 
```
W 10/24 10:07:06 /apis/application.giantswarm.io/v1alpha1/namespaces/giantswarm/charts/bad-url releasev1.GetDesiredState retrying due to error | operatorkit/resource/wrapper/retryresource/crud_resource_ops_wrapper.go:96 | controller=chart-operator-chart | event=update | loop=3 | underlyingResource=releasev1 | version=1305
	/Users/jihyukbok/go/src/github.com/giantswarm/chart-operator/vendor/github.com/giantswarm/operatorkit/resource/wrapper/retryresource/crud_resource_ops_wrapper.go:89:
	/Users/jihyukbok/go/src/github.com/giantswarm/chart-operator/service/controller/chart/v1/resource/release/desired.go:30:
	/Users/jihyukbok/go/src/github.com/giantswarm/chart-operator/vendor/github.com/giantswarm/helmclient/pull_chart_tarball.go:26:
	/Users/jihyukbok/go/src/github.com/giantswarm/chart-operator/vendor/github.com/giantswarm/helmclient/pull_chart_tarball.go:79:
	/Users/jihyukbok/go/src/github.com/giantswarm/chart-operator/vendor/github.com/giantswarm/backoff/retry.go:23:
	/Users/jihyukbok/go/src/github.com/giantswarm/chart-operator/vendor/github.com/giantswarm/helmclient/pull_chart_tarball.go:40:
	Get https://just-a-bad-url.nothing/non-existent.tgz: dial tcp: lookup just-a-bad-url.nothing: no such host
```

Reply:
Would it make sense just grep the error string to find out whether this is no host issue? I think network error types vary so it doesn't matter to retry at this moment.

Reply:
Yes I think this is too generic as is.

What about an internal error matcher for no such host? So we only use backoff.Permanent then.